components/org.wso2.carbon.appmgt.gateway/src/main/java/org/wso2/carbon/appmgt/gateway/internal/AppManagerGatewayComponent.java

```diff
@@ -84,20 +60,32 @@ protected void deactivate(ComponentContext componentContext) {
         }
     }
 
-    protected void setAPIManagerConfigurationService(AppManagerConfigurationService amcService) {
+    protected void setAppManagerConfigurationService(AppManagerConfigurationService amcService) {
         if (log.isDebugEnabled()) {
-            log.debug("Gateway manager configuration service bound to the WebApp host objects");
+            log.debug("App manager configuration service is set to gateway bundle");
         }
-        configuration = amcService.getAPIManagerConfiguration();
-        org.wso2.carbon.appmgt.impl.service.ServiceReferenceHolder.getInstance().setAPIManagerConfigurationService(amcService);
+        ServiceReferenceHolder.getInstance().setAPIManagerConfigurationService(amcService);
     }
 
-    protected void unsetAPIManagerConfigurationService(AppManagerConfigurationService amcService) {
+    protected void unsetAppManagerConfigurationService(AppManagerConfigurationService amcService) {
         if (log.isDebugEnabled()) {
-            log.debug("Gateway manager configuration service unbound from the WebApp host objects");
+            log.debug("App manager configuration service is unset to gateway bundle");
```

Comment:
Fixed in https://github.com/wso2/carbon-appmgt/pull/539/commits/05d5cef1f9e3d525eb6d0c0af34943c96daac8ae